### PR TITLE
Remove extra quotes for strings in metadata formatter

### PR DESCRIPTION
**Pull request by itamarkam**
Commit Message: Remove extra quotes for strings in metadata formatter
Additional Description: string values in metadata are handled the same way objects are, which adds escaped quotes around  them. This change removes these extra quotes (with a runtime flag "unquote_log_string_values").
Risk Level: Low - will change logging behavior for string metadata values, but flag protected.
Testing: Unit tests.
Docs Changes: N/A
Release Notes: access_log: remove extra quotes on metadata string values. This behavior can be temporarily reverted by setting ``envoy.reloadable_features.unquote_log_string_values`` to false.
Platform Specific Features:
[Optional Runtime guard:] unquote_log_string_values runtime flag (defaults to new behavior)
[Optional Fixes #Issue]
[Optional Deprecated:]
[Optional [API Considerations](https://github.com/envoyproxy/envoy/blob/main/api/review_checklist.md):]

### CODE REVIEW

**Review by yanavlasov — DISMISSED** [review]

**Comment by yanavlasov** [comment]
LGTM, the format complains about the ordering of the runtime keys in the array. Good to go after it is fixed.

/wait

**Comment by itamarkam** [comment]
> LGTM, the format complains about the ordering of the runtime keys in the array. Good to go after it is fixed.
> 
> /wait

Done.

**Review by yanavlasov — APPROVED** [review]

**Comment by yanavlasov** [comment]
It looks like //test/integration:proxy_proto_integration_test uses access logging in tests and expects double quotes. I think we can just fix the test to use single quotes, since it is the new expectation.

**Review by yanavlasov — DISMISSED** [review]

**Comment by yanavlasov** [comment]
We may need to wait for windows build to be fixed and merge main.

**Comment by yanavlasov** [comment]
Windows build was fixed. Please merge main.

**Comment by itamarkam** [comment]
> Windows build was fixed. Please merge main.

Done.

**Review by yanavlasov — APPROVED** [review]

**Pull request merged**
